dev/how-to/maintain/tidb-in-kubernetes/log-collecting.md

```diff
@@ -0,0 +1,83 @@
+---
+title: 日志收集
+category: how-to
+---
+
+# 日志的收集
+
+系统与程序的运行日志对排查问题以及实现一些自动化操作可能非常有用。本文将简要说明收集 TiDB 及相关组件日志的方法。
+
+## TiDB 与 Kubernetes 组件运行日志
+
+通过 TiDB Operator 部署的 TiDB 各组件默认将日志输出在容器的 `stdout` 和 `stderr` 中，对于 Kubernetes 而言，这些日志会被存放在宿主机的 `/var/log/containers` 目录下，并且文件名中包含了 Pod 和容器名称等信息。故而对容器中应用的日志收集可以直接在宿主机上完成。
+
+如果在你的现有基础设施中已经有用于收集日志的系统，只需要通过常规方法将 Kubernetes 所在的宿主机上的 `/var/log/containers/*.log` 文件加入采集范围即可；如果没有可用的日志收集系统，或者希望部署一套独立的系统用于收集相关日志，也可以使用你熟悉的任意日志收集系统或方案。
+
+Kubernetes 官方文档中提供了 [ElasticSerach](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-elasticsearch-kibana/) 和 [Stackdriver](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-stackdriver/) 两种日志收集方案可供参考。
+
+常见的可用于收集 Kubernetes 日志的开源工具有：
+ - Filebeat
+ - Filebit
+ - Fluentd
+ - Fluent-bit
+ - ...
+
+收集到的日志通常可以汇总存储在某一特定的服务器上，或存放到 ElasticSerach 等专用的存储、分析系统当中。
+
+一些云服务商或专门的性能监控服务提供商也有各自的免费或收费的日志收集方案可以选择。
+
+如果不通过单独的日志收集工具汇总日志，你也可以直接使用 `kubectl` 工具查看某个容器的运行日志，这一方法无法查看已销毁容器的日志或者容器重启之前的日志：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl logs -n ${namespace} ${tidbPodName}
+```
+
+如果需要从多个 Pod 获取日志，可以使用 [`stern`](https://github.com/wercker/stern).
+
+{{< copyable "shell-regular" >}}
+
+```shell
+stern -n ${namespace} tidb -c slowlog
+```
+
+## TiDB 慢查询日志
+
+默认情况下，TiDB 会打印慢查询日志到标准输出，和正常日志混在一起。你可以通过关键字 `SLOW_QUERY` 来筛选慢查询日志，例如：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl logs -n ${namespace} ${tidbPodName} | grep SLOW_QUERY
+```
+
+在一些情况下，你可能希望使用一些工具或自动化系统对日志内容进行解析。TiDB 各组件的应用日志使用了[统一的日志格式](https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md)所以解析通常不是问题，但由于慢查询日志使用了与 MySQL 兼容的多行格式，与应用日志混在一起时可能会对解析造成困难。
+
+在 `values.yaml` 文件是配置 `separateSlowLog` 参数可以将慢查询日志输出到一个专用的旁路容器中，这样慢查询日志在宿主机上会被输出到一个单独的文件，和应用日志分开。
```

Comment:
`separateSlowLog` is now enabled by default, see https://github.com/pingcap/tidb-operator/pull/610

Reply:
Yes, besides TiDB 3.0 would not emit slow log in the stdout/stderr.

Reply:
So what if commenting out this line in `values.yaml`? Will it been disabled again as former default, or user has to set it to `false`?